src/test/java/io/confluent/connect/elasticsearch/bulk/BulkProcessorTest.java

```diff
@@ -268,4 +278,54 @@ public void unretriableErrors() throws InterruptedException {
     }
   }
 
+  @Test
+  public void computeRetryBackoffForValidRanges() {
+    assertComputeRetryInRange(1, 10L, 10000L);
+    assertComputeRetryInRange(10, 10L, 10000L);
+    assertComputeRetryInRange(100, 10L, 10000L);
+
+    assertComputeRetryInRange(1, 100L, 1000L);
+    assertComputeRetryInRange(10, 100L, 1000L);
+    assertComputeRetryInRange(100, 100L, 1000L);
+
+    assertComputeRetryInRange(1, 100L, 100L);
+    assertComputeRetryInRange(10, 100L, 100L);
+    assertComputeRetryInRange(10, 100L, 101L);
+    assertComputeRetryInRange(100, 100L, 100L);
+  }
+
+  @Test
+  public void computeRetryBackoffForNegativeBackoffTime() {
+    assertComputeRetryInRange(1, -100L, 1000L);
+    assertComputeRetryInRange(10, -100L, 1000L);
+    assertComputeRetryInRange(100, -100L, 1000L);
+
+    assertComputeRetryInRange(1, 100L, -10L);
+    assertComputeRetryInRange(10, 100L, -10L);
+    assertComputeRetryInRange(100, 100L, -10L);
+
+    assertComputeRetryInRange(1, 100L, -1000L);
+    assertComputeRetryInRange(10, 100L, -1000L);
+    assertComputeRetryInRange(100, 100L, -1000L);
+  }
+
+  @Test
+  public void computeRetryBackoffForInvalidRanges() {
+    long minRetryBackoffMs = 100L;
+    long maxRetryBackoffMs = 10L;
+    for (int retryAttempt=1; retryAttempt < 100; ++retryAttempt) {
+      long result = BulkProcessor.computeRetryWaitTimeInMillis(retryAttempt, minRetryBackoffMs, maxRetryBackoffMs);
+      assertTrue(result >= maxRetryBackoffMs);
+    }
+  }
+
+  protected void assertComputeRetryInRange(int retryAttempts, long minRetryBackoffMs, long maxRetryBackoffMs) {
+    for (int retryAttempt=1; retryAttempt < retryAttempts; ++retryAttempt) {
+      long result = BulkProcessor.computeRetryWaitTimeInMillis(retryAttempt, minRetryBackoffMs, maxRetryBackoffMs);
+      if (maxRetryBackoffMs > 0) assertTrue(result <= maxRetryBackoffMs);
+      assertTrue(result >= minRetryBackoffMs);
```

Comment:
I believe it's somewhere here that we should check that the wait is truly exponential.